config/config.go

```diff
@@ -59,6 +66,11 @@ func (cfg *Config) userConfigLoaded() bool {
 	return cfg.userConfigLocation != ""
 }
 
+// EnableEventPublishing enables config event publishing to the event bus
+func (cfg *Config) EnableEventPublishing(eb eventbus.EventBus) {
+	cfg.eventBus = eb
```

Comment:
why though? why not just inject this as a dependency and no have the nil check in set user?

Reply:
Not nil check is just a safeguard. This essentially is the dependency injection, because `config.Current` is a one global instance.